### PR TITLE
Fixed the problem of generating wrong id

### PR DESCRIPTION
**Pull request by Qanora**
[description]
when i typed 1000,
I got n1030_minimum_cost_to_merge_stones.rs
[fixed]
Incorrect use tag "problem.stat.frontend_question_id" vs "problem.stat.question_id"

### CODE REVIEW

**Comment by mapx** [comment]
Leetcode problems have both "frontend id" and "backend id". All problems in the repo are indexed with their "backend id". The framework accepts "frontend id" to generate the problem file (with backend id).

It's a feature by design, for some reason.

File: src/problem.rs [hunk 1]
```diff
@@ -42,7 +42,7 @@ pub fn get_problem(frontend_question_id: u32) -> Option<Problem> {
                 content: resp.data.question.content,
                 sample_test_case: resp.data.question.sample_test_case,
                 difficulty: problem.difficulty.to_string(),
-                question_id: problem.stat.question_id,
+                question_id: problem.stat.frontend_question_id,
```
Comment (Qanora) [inline src/problem.rs:45]: Judging by problem.stat.**frontend_question_id** at line 23
Generating from question_id: problem.stat.**question_id** at line 45

**Comment by Qanora** [comment]
> Leetcode problems have both "frontend id" and "backend id". All problems in the repo are indexed with their "backend id". The framework accepts "frontend id" to generate the problem file (with backend id).
> 
> It's a feature by design, for some reason.

I am not quite sure why it is a feature ( indexed by front id but generated by backend id)

**Comment by aylei** [comment]
@Qanora I have not noticed that the `frontend_question_id` may differ from the `question_id`, I just assumed that they would always be same previously.

Here is my conclusion after looking at the response of leetcode API:

The `frontend_question_id` is the ordinal displayed in the problem list:

image

While `question_id` seems to be an unique identifier of an question. And seems that there is no strong relation between these two ids (the `frontend_question_id` of question id `1049` is `999`). Because most of the tutorials and blogs refers to leetcode questions by `frontend_question_id` (e.g. "`Solution of leetcode #1000`"), I think generate the question file with `frontend_question_id` would be better.

**Comment by Qanora** [comment]
I agree with you, so is this ready to merge? @aylei

**Comment by mapx** [comment]
All existing problem files need id remapping for consistency as well, with respect to this update.

**Review by aylei — APPROVED** [review]
LGTM

**Comment by aylei** [comment]
Thanks! @Qanora

**Comment by mapx** [comment]
I found `frontend_question_id` not as stable as `question_id`.

When you generated problem `1344`, you got `n1344_jump_game_v.rs` a few days ago, but it's `n1344_angle_between_hands_of_a_clock.rs` now.

Confusing...

**Comment by Qanora** [comment]
> I found `frontend_question_id` not as stable as `question_id`.
> 
> When you generated problem `1344`, you got `n1344_jump_game_v.rs` a few days ago, but it's `n1344_angle_between_hands_of_a_clock.rs` now.
> 
> Confusing...

I tried to generate problem ```1344``` on this commit and the latest master branch, i got ```p1344_angle_between_hands_of_a_clock.rs```. It looks like fine.

**Comment by mapx** [comment]
> I tried to generate problem 1344 on this commit and the latest master branch, i got p1344_angle_between_hands_of_a_clock.rs. It looks like fine.

It's the fact that leetcode team sometimes shifts front-end id. "Jump Game V" was 1344 and is 1340. IMO, current PR has no fault. There is no easy way to deal with that.

**Pull request merged**